enterprise/core-edge/src/test/java/org/neo4j/coreedge/scenarios/CoreToCoreCopySnapshotIT.java

```diff
@@ -251,4 +253,9 @@ private String string( int numberOfCharacters )
         }
         return s.toString();
     }
+
+    private CoreState getCoreState( CoreGraphDatabase follower )
+    {
+        return follower.getDependencyResolver().resolveDependency( CoreState.class );
```

Comment:
The implementation looked horrible, and would love to see a better one. And sure they were only used in tests because they in fact formed part of the surface (CoreGraphDatabase, not necessarily meant for general public consumption), so being used by an IT is the highest level you could observe internally. I don't know about the circular dependency though, sounds like an issue with the facades, factories, and whatnot. I wish I understood that design...